### DISCUSSION TITLE
How to specify via a property additional arguments for an argument list?

### DISCUSSION THREAD

**Discussion by merks** [eclipse-tycho/tycho#1824] [Q&A] [answered]
Previously one could specify an argument line to run an Eclipse application:

https://tycho.eclipseprojects.io/doc/2.7.5/tycho-extras/tycho-eclipserun-plugin/eclipse-run-mojo.html#appArgLine

Now you **must** use a list:

https://tycho.eclipseprojects.io/doc/3.0.0/tycho-extras/tycho-eclipserun-plugin/eclipse-run-mojo.html#applicationArgs

This is arguably more well structured, but also very verbose.  I.e., this 

![image](https://user-images.githubusercontent.com/208716/206120425-325cdf1d-063f-44d0-a7cc-ca9d480a0bd5.png)

versus this

![image](https://user-images.githubusercontent.com/208716/206120568-c9793142-0e5e-4e86-b254-79aaf810aa40.png)

Okay, fine, not so readable, harder to edit, but now I wonder how can I add to the arguments via a property like I could previously?

Previously I could specify this property this:
`
<org.eclipse.justj.p2.manager.args>-remote ${org.eclipse.storage.user}@projects-storage.eclipse.org:/home/data/httpd/download.eclipse.org/${org.eclipse.download.location.relative}</org.eclipse.justj.p2.manager.args>
`
to specify two additional arguments and have it inserted in the argument line.  But how to do that with this new format?  My question is similar to this:

https://stackoverflow.com/questions/24170098/maven-property-referencing-a-list

I.e., I see no way to pass additional arguments (and/or even optional arguments) using this list of `<arg>` design.   Perhaps I've overlooked a way because otherwise the problem of "mangled argument processing" has to be pushed down to the receiving application's argument processing logic  (e.g., "-foo bar" as a single argument is treated like two arguments and perhaps "" is discarded), but that kind of additional logic might not be within my control if it's not my application being launched.  So while the design my be more accurate and well-formed, it seems to exclude capabilities previously possible.  Again, hopefully I've missed some tricky way to accomplish the same thing.

**Comment by merks** [comment, marked as answer]
Even surefire retains the ability to specify argline which could be used to specify additional system properties and of course the vm args

- https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#argLine
- https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#systemPropertyVariables

But now there is only this for the JVM args also making that more difficult (impossible?) to specialize via property expansion:

https://tycho.eclipseprojects.io/doc/3.0.0/tycho-extras/tycho-eclipserun-plugin/eclipse-run-mojo.html#jvmArgs

Tycho surefire still works the "old" way:

https://tycho.eclipseprojects.io/doc/latest/tycho-surefire-plugin/test-mojo.html

This makes it seem more odd and kind of questionable...

**Reply by mickaelistria** [reply]
I guess the change was too eagerly assumed as a cleanup and simplification. I think Tycho should mimic surefire here. PRs towards would most likely be welcome.

**Reply by laeubi** [reply]
I don't see a reason not having an `argLine` (that is a space separated "list" and "applicationArgs" that is a list, the code just needs to check what one is possibly set and probably merge both ... I think the main reason was not "cleanup" but problem with arguments that contain spaces.

**Comment by merks** [comment]
The argLine and appArgLine options have been restored for Tycho 3.0.2 and for 4.x.
